cmd/restic/cmd_restore.go

```diff
@@ -176,6 +184,31 @@ func runRestore(ctx context.Context, opts RestoreOptions, gopts GlobalOptions,
 
 	excludePatterns := filter.ParsePatterns(opts.Exclude)
 	insensitiveExcludePatterns := filter.ParsePatterns(opts.InsensitiveExclude)
+
+	if len(opts.ExcludeFiles) > 0 {
```

Comment:
At least the exclude patterns look like we could share the implementation used by the `backup` command. Please remove the custom exclude pattern implementation from restore and use `excludePatternOptions` instead. Unless I've missed something, the returned `RejectByNameFunc`s should suitable for use in selectExcludeFilter.

Reply:
This is done.